packages/snap/src/state.ts

```diff
@@ -0,0 +1,367 @@
+      method: "snap_manageState",
+      params: { operation: "get" },
+    });
+
+    // parse the JSON data into a plain object
+    let parsedData = JSON.parse(data?.chains?.toString()!);
+
+    // create a new Chains object and populate it with the parsed data
+    let chains = new Chains(parsedData);
+
+    // add the chain into chains class
+    chains.addChain(chain);
+
+    // update Metamask state with new chain state
+    await snap.request({
+      method: "snap_manageState",
+      params: { operation: "update", newState: { ...data, chains: chains.string() } },
+    });
+
+    return chains;
+  }
+  /**
+   * Adds a list of Chains as state into Metamask snap.
+   *
+   * WARNING: This replaces the entire state within Metamask so any chains in state will be replaced.
+   *
+   * @param chains Chains to replace entire Metamask chain state with.
+   * @returns Returns the current state of Chains.
+   * @throws If an error occurs.
+   */
+  public static async addChains(chains: Chains): Promise<Chains> {
+    // get current state
+    const data = await snap.request({
+      method: "snap_manageState",
+      params: { operation: "get" },
+    });
+    // update Metamask state with new chain state
+    await snap.request({
+      method: "snap_manageState",
+      params: {
+        operation: "update",
+        newState: { ...data, chains: chains.string() },
+      },
+    });
+    return chains;
+  }
+  public static async removeChain(chain_id: string): Promise<Chains> {
+    // get the current state of chains in Metamask we will add chain into
+    const data = await snap.request({
+      method: "snap_manageState",
+      params: { operation: "get" },
+    });
+    if (data?.chains == undefined || data?.chains == null) {
+      throw new Error("Snap has not been initialized. Please initialize snap.");
+    }
+    // remember we keep chain stores as a json string so convert into a Chains object
+    let chains = new Chains(JSON.parse(data?.chains?.toString()!));
+    let keepChains = chains.chains.filter((item) => item.chain_id !== chain_id);
+    // update chains
+    chains.chains = keepChains;
+
+    // update Metamask state with new chain state
+    await snap.request({
+      method: "snap_manageState",
+      params: { operation: "update", newState: { ...data, chains: chains.string() } },
+    });
+
+    return chains;
+  }
+}
+
+/**
+ * AddressState is the class to manage all address book state within Metamask.
+ */
+export class AddressState {
+  /**
+   * Gets all Cosmos addresses from Metamask snap state.
+   *
+   * @returns All Addresses in Metamask snap state.
+   * @throws If an error occurs.
+   */
+
+  public static async getAddressBook(): Promise<Addresses> {
+    // get the current state of addresses in Metamask
+    const data = await snap.request({
+      method: "snap_manageState",
+      params: { operation: "get" },
+    });
+
+    if (data?.addresses == undefined || data?.addresses == null) {
```

Comment:
Please replace all occurences of this pattern.

```suggestion
    if (data?.addresses == null) {
```